### ISSUE TITLE
Export without acceding application

### ISSUE THREAD

**Issue by Aloha911** [SilenceIM/Silence#398]
Hello,

I broke the glass of my phone this weekend and I would like to export my SMS regardless. I got an old phone on which I installed the application but I find no way to import my existing SMS.

I was not able to perform backups of existing SMS since my phone is unusable now, but I still have access to the ROM of the broken phone through my computer.

I tried to copy the content of /data/data/org.smssecure.smssecure directory on my new (rooted) phone, but the application crashes when I try to start it then. If I restart the phone, it believes that I just installed the application.

Is there a way to do what I want?

Sorry for my english.
Thank you.

**Comment by pR0Ps** [comment]
The file "libcurve25519.so" is platform-specific and shouldn't be copied
over. Try installing Silence on the new phone, force stopping it and any
services it starts, then copying over everything in the
"/data/data/org.smssecure.smssecure"
directory *except* the "lib" folder (IIRC it's a symlink) with
"libcurve25519.so"
in it.

At this point opening the app should leave you in the exact same state as
your previous phone. Note that you may have to re-initiate secure sessions
with your contacts.

**Comment by Aloha911** [comment]
Yeah, i got it !

After copying files except lib link, i still got an error. I checked with logcat for more details, and there was a permissions issue on Silence files. I just did a chown on all application files with correct user (u0_a45 on my phone). After that, i get all my SMS back.

There was a last issue. I use a passphrase to protect access to messages, but application preferences said that I didn't have any passphrase set and didn't allow me to change it. To correct that, I change **org.smssecure.smssecure_preferences.xml** in the shared_prefs folder :
<boolean name="pref_disable_passphrases" value="false" />
<boolean name="pref_enable_passphrase_temporary" value="true" />

After rebooting the phone, I was able to change my passphrase and access all my SMS and MMS :)
I just lost secure sessions like you said but it's really not a problem.

Again, sorry for my english, and lot a thanks for your help.
Good luck for the future! :-)
